src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj

```diff
@@ -92,7 +92,7 @@
     <PathToMsalSources>$(MSBuildThisFileDirectory)</PathToMsalSources>
     <!-- Need to include / exclude / remove cs files manually so that they are present in both assemblies -->
     <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
-    <NoWarn>$(NoWarn);CS8002;NU5131;NETSDK1138</NoWarn>
+    <NoWarn>$(NoWarn);CS8002;NU5131;NETSDK1138;SYSLIB0026;SYSLIB0027;SYSLIB0028</NoWarn>
```

Comment:
Even with `JsonIgnore` on that X509Certificate2, it looks like SJT is issuing warnings about invalid use of obsolete properties from X509Certificate2 (e.g. PublicKey, PrivateKey ... full list here https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/obsoletions-overview). We don't use these properties. 

So I disabled the warnings. I'm on the fence about this one, as disabling the warning may cause MSAL developers to use a deprecated x509cert property.

Reply:
Can't we do `#pragma warning disable` and then enable it instead?

Reply:
I agree with Neha, lets add `#pragma warning disable` or an entry to global suppressions instead so we can have a more targeted approach. Also, we have .editorconfig, global changes like this go in there right?

Reply:
Pragma doesn't work since this happens in the generated files. Not sure how to disable only for STJ stuff and why JsonIgnore doesn't work. But yea, I wouldn't disable these warnings for everything.

Reply:
Found a related conversation in STJ repo: https://github.com/dotnet/runtime/issues/76829#issuecomment-1349391310